### ISSUE TITLE
[🐛BUG] Discrepancies in Pop model metric values with different train_batch_size

### ISSUE THREAD

**Issue by anruijian** [RUCAIBox/RecBole#1844]
**Describe the bug**
The Pop model metric values differ a lot with different `train_batch_size`.

**To Reproduce**
```python
parameter_dict = {
    'train_batch_size': 1024,
    "eval_args": {"split": {"LS": "valid_and_test"}, "order": "TO", "mode": "full", "group_by": "user"},
    "show_progress": False,
}
run_recbole(model='Pop', dataset='ml-100k', config_dict=parameter_dict)
"""
Output with train_batch_size = 1024:
{'best_valid_score': 0.0245,
 'valid_score_bigger': True,
 'best_valid_result': OrderedDict([('recall@10', 0.0721),
              ('mrr@10', 0.0245),
              ('ndcg@10', 0.0355),
              ('hit@10', 0.0721),
              ('precision@10', 0.0072)]),
 'test_result': OrderedDict([('recall@10', 0.0859),
              ('mrr@10', 0.0306),
              ('ndcg@10', 0.0434),
              ('hit@10', 0.0859),
              ('precision@10', 0.0086)])}
"""
parameter_dict = {
    'train_batch_size': 8192,
    "eval_args": {"split": {"LS": "valid_and_test"}, "order": "TO", "mode": "full", "group_by": "user"},
    "show_progress": False
}
run_recbole(model='Pop', dataset='ml-100k', config_dict=parameter_dict)
"""
Output with train_batch_size = 8192:
{'best_valid_score': 0.0043,
 'valid_score_bigger': True,
 'best_valid_result': OrderedDict([('recall@10', 0.0212),
              ('mrr@10', 0.0043),
              ('ndcg@10', 0.0081),
              ('hit@10', 0.0212),
              ('precision@10', 0.0021)]),
 'test_result': OrderedDict([('recall@10', 0.0244),
              ('mrr@10', 0.0065),
              ('ndcg@10', 0.0104),
              ('hit@10', 0.0244),
              ('precision@10', 0.0024)])}
"""
```
Furthermore, upon reloading the saved model checkpoints, the `item_cnt` and `max_cnt` values differ between the two models.
```
model.item_cnt
"""
tensor([[  0],
        [106],
        [166],
        ...,
        [ 55],
        [ 57],
        [ 52]], device='cuda:0')
"""
model2.item_cnt
"""
tensor([[ 0],
        [24],
        [24],
        ...,
        [22],
        [23],
        [22]], device='cuda:0')
"""
```

**Expected behavior**
I believe the Pop model only calcualtes the count of an item in the training set, which should be the same regardless of the train_batch_size/ 

**Screenshots**

**Colab Links**
If applicable, add links to Colab or other Jupyter laboratory platforms that can reproduce the bug.

**Desktop (please complete the following information):**
 - OS: [e.g. Linux, macOS or Windows]
- RecBole Version [1.1.1]
 - Python Version [3.10.11]
- PyTorch Version [1.13.1+cu117]

**Comment by chenyuwuxin** [comment]
@anruijian Thanks for your attention to RecBole!
Thank you very much for your feedback. After our investigation, it appears that some items may be double-counted due to different batch sizes. To address this issue, you will need to set a very large batch size (e.g., 10,000,000). Of course, we will work on optimizing this problem in future updates. We sincerely apologize for any inconvenience this may have caused you.
